src/ConcurrentIteratorChannel.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Amp\Sync;
+
+use Amp\ByteStream\ClosableStream;
+use Amp\Cancellation;
+use Amp\Pipeline\ConcurrentIterator;
+use Amp\Pipeline\Queue;
+
+/**
+ * Creates a channel from a Pipeline and Emitter. The Pipeline emits data to be received on the channel (data
+ * emitted on the Pipeline will be returned from calls to {@see Channel::receive()}). The Emitter will receive data
+ * that sent on the channel (data passed to {@see Channel::send()} will be passed to {@see Emitter::yield()}).
+ *
+ * @template TReceive
+ * @template TSend
+ * @template-implements Channel<TReceive, TSend>
+ */
+final class ConcurrentIteratorChannel implements Channel, ClosableStream
```

Comment:
Do we need this class outside of `LocalChannel`? Can this be rolled directly into `LocalChannel`?

Reply:
This is used by `createChannelPair()`. I'm not sure how useful `LocalChannel` even is outside of test cases, where the pair could be used instead. Thinking `LocalChannel` should be removed to avoid misuse.

Reply:
`LocalChannel` can be used to synchronize two coroutines with message passing, no?

Reply:
No, it can't be. You need to generate a pair to do that.

Reply:
Well… unless one coroutine strictly sends and the other strictly receives, but you can also do that with a pair and with less confusion for end users.